tests/v2_validation/cattlevalidationtest/core/common_fixtures.py

```diff
@@ -3362,3 +3363,69 @@ def wait_for_state(self, obj, state):
         obj = self.client.wait_success(obj)
         assert obj.state == state
         return obj
+
+
+@pytest.fixture(scope='session')
+def deploy_catalog_template(client, super_client, request,
+                            t_name, t_version, t_env):
+    catalog_url = rancher_server_url() + "/v1-catalog/templates/community:"
+    # Deploy Catalog template from catalog
+    r = requests.get(catalog_url + t_name + ":" + str(t_version))
+    template = json.loads(r.content)
+    r.close()
+
+    dockerCompose = template["files"]["docker-compose.yml"]
+    rancherCompose = template["files"]["rancher-compose.yml"]
+
+    environment = t_env
+    env = client.create_stack(name=t_name,
+                              dockerCompose=dockerCompose,
+                              rancherCompose=rancherCompose,
+                              environment=environment,
+                              startOnCreate=True)
+    env = client.wait_success(env, timeout=300)
+    for i in range(10):
+        testenv = client.list_stack().data[0]
+        if testenv.healthState == "unhealthy":
+            time.sleep(1)
+            continue
+        assert (testenv.state == "active" or
+                testenv.state == "activating")
+        assert (testenv.healthState == 'healthy' or
+                testenv.healthState == "initializing")
+        time.sleep(1)
+    assert (testenv.state == "active" or
+            testenv.state == "activating")
+    assert (testenv.healthState == 'healthy' or
+            testenv.healthState == "initializing")
+
```

Comment:
@galal-hussein can we use wait_for_condition() to wait for environment to get to requited state and health state?